docs/writing_a_parser_plugin/parser_plugin_overview.md

```diff
@@ -0,0 +1,75 @@
+for individual parsers are contained within a corresponding subdirectory: `<parserproject>parsers/<parsername>`
+For example, the Quantum Espresso parser files are found in `dependencies/parsers/electronic/electronicparsers/quantumespresso/`.
+
+
+### Setting up your development branches
+
+We will first focus on the case of adding a new parser to an existing parser project.
+[Creating a new parser project](references/advanced_new_parser_project.md) will require a few extra steps.
+The existing parser projects are stored within their own git repositories and then linked
+to the NOMAD software. All current parser projects are available at [nomad-coe](https://github.com/nomad-coe)
+(see also individual links above).
+
+You will first need to create new branches within *both* the NOMAD project and *also* within the corresponding
+parser project. Ideally, this should be done following the [best practices for NOMAD development](references/quick_NOMAD_best_practices.md).
+Here, we briefly outline the procedure:
+
+Create a new issue within the NOMAD project at [NOMAD gitlab](https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR.git).
+On the page of the new issue, in the top right, click the arrow next to the `Create merge request`
+button and select `Create branch`. The branch name should be automatically generated with the
+corresponding issue number and the title of the issue (copy the branch name to the clipboard for use below),
+and the default source branch should be `develop`.
+Click the `Create branch` button.
+
+Now, run the following commands in your local NOMAD directory:
+
+`git fetch --all` &nbsp;&nbsp;&nbsp;&nbsp; (to sync with remote)
+
+`git checkout origin/<new_branch_name> -b <new_branch_name>` &nbsp;&nbsp;&nbsp;&nbsp; (to checkout the new branch and create a local copy of the branch)
+
+Unless you *just* installed the NOMAD development version, you should rerun `./scripts/setup_dev_env.sh`
+within the NOMAD directory to reinstall with the newest development branch.
+
+Now we need to repeat this process for the parser project that we plan to extend. As above,
+create a new issue at the relevant parser project GitHub page. Using the identical
+issue title as you did for the NOMAD project above is ideal for clarity.
+On the page of the new issue, in the right sidebar under the subsection `Development`, click
+`Create a branch`. As above, the branch name should be automatically generated with the
+corresponding issue number and the title of the issue, and the default source branch should be `develop`
+(which can be seen by clicking `change source branch`).
+Under `What's next`, the default option should be `Checkout locally`, which is what we want in this case.
+Click `Create branch`, and then copy the provided commands to the clipboard, and run them within the parser
+project folder within your local NOMAD repo, i.e., `dependencies/parsers/<parserproject>`.
```

Comment:
This all falls under basic git usage... Isn't there a git tutorial we could refer to?

Reply:
@ndaelman-hu I agree with you in principle...however, there are many ways to use git, and working on a larger software project is not the same as using git for your personal repos or within a small group. The idea here is that we guide the to-be developer according to the standards of git usage within the NOMAD project. Moreover, these docs should be useful for not only providing guidance to external users, but also to train newcomers to the internal NOMAD team.

I do think probably we should move this to a reference (I will make a TODO), so it is not crowding the more important content, but I would like some sort of clear description of how the development process works within the NOMAD git project.